### PR TITLE
Fixes #562 adds Windows demo setup help

### LINKED ISSUE #562: Cannot run demo on Windows
npm start is failing on Windows 10.

## Details
- Browser and browser version: ???
- OS version: Windows 10
- xterm.js version: 2.2.3

### Steps to reproduce

1. Use Windows 10
2. Open PowerShell
3. npm install
4. npm run

```

=> npm install

> xterm@2.2.3 prepublish D:\xterm.js
> npm run build


> xterm@2.2.3 build D:\xterm.js
> gulp build

[20:41:33] Using gulpfile D:\xterm.js\gulpfile.js
[20:41:33] Starting 'tsc'...
[20:41:36] Finished 'tsc' after 3.51 s
[20:41:36] Starting 'browserify'...
[20:41:39] Finished 'browserify' after 2.67 s
[20:41:39] Starting 'sorcery'...
[20:41:39] Finished 'sorcery' after 172 ms
[20:41:39] Starting 'build'...
[20:41:39] Finished 'build' after 5.9 μs
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules\chokidar\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.1: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

=> npm start

> xterm@2.2.3 prestart D:\xterm.js
> npm run build


> xterm@2.2.3 build D:\xterm.js
> gulp build

[20:41:55] Using gulpfile D:\xterm.js\gulpfile.js
[20:41:55] Starting 'tsc'...
[20:41:59] Finished 'tsc' after 3.45 s
[20:41:59] Starting 'browserify'...
[20:42:02] Finished 'browserify' after 2.77 s
[20:42:02] Starting 'sorcery'...
[20:42:02] Finished 'sorcery' after 171 ms
[20:42:02] Starting 'build'...
[20:42:02] Finished 'build' after 5.59 μs

> xterm@2.2.3 start D:\xterm.js
> node demo/app

Cannot find module '..\build\Release\pty.node'
module.js:457
    throw err;
    ^

Error: Cannot find module '..\bin\win32\x64.m48.node'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Module.require (module.js:483:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (D:\xterm.js\node_modules\node-pty\lib\pty.js:18:9)
    at Module._compile (module.js:556:32)
    at Object.Module._extensions..js (module.js:565:10)
    at Module.load (module.js:473:32)
    at tryModuleLoad (module.js:432:12)
    at Function.Module._load (module.js:424:3)

npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\brian\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "start"
npm ERR! node v6.5.0
npm ERR! npm  v3.10.8
npm ERR! code ELIFECYCLE
npm ERR! xterm@2.2.3 start: `node demo/app`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the xterm@2.2.3 start script 'node demo/app'.
npm ERR! Make sure you have the latest version of node.js and npm installed. npm ERR! If you do, this is most likely a problem with the xterm package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node demo/app
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs xterm
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls xterm
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     D:\xterm.js\npm-debug.log

=>
```

### PR DESCRIPTION
**Pull request by wavebeem**

### CODE REVIEW

**Comment by wavebeem** [comment]
It looks like the tests failed, but all I changed was the README, so I assume I didn't cause that?

File: README.md [hunk 1]
```diff
@@ -48,14 +48,30 @@ Xterm.js works seamlessly in Electron apps and may even work on earlier versions
...
+
+Windows requires an additional dependency to run the demo. Also, do not use ConEmu, as it seems to break the demo.
+
+First, ensure [node-gyp](https://github.com/nodejs/node-gyp) is installed and configured correctly, then run these commands.
 
 ```
-npm install
-npm start
+> npm install
+> npm install node-pty
```
Comment (Tyriar) [inline README.md]: You shouldn't need to do this as it's done in `npm install`

**Comment by wavebeem** [comment]
Oh, I see, I think that was just because my build stuff wasn't set up right, and that was the only module I needed to reinstall. Removing.

**Comment by Tyriar** [comment]
The test failure is related to TypeScript 2.2, I made a commit to fix the issue.

Thanks @wavebeem :smile:

**Comment by wavebeem** [comment]
Thanks for your help and the sweet project! 👍

**Pull request merged**